pkg/ccl/backupccl/system_schema.go

```diff
@@ -103,6 +113,114 @@ func jobsRestoreFunc(
 ) error {
 	executor := execCfg.InternalExecutor
 
+	const (
+		statesToRevert = `('` + string(jobs.StatusRunning) + `', ` +
+			`'` + string(jobs.StatusPauseRequested) + `', ` +
+			`'` + string(jobs.StatusPaused) + `')`
+	)
+
+	err := func() (err error) {
+		// makeJobsList writes the given jobs as a SQL tuple set to the
```

Comment:
Is there something that has prevented these jobs from being adopted?

Reply:
This updating is being done in a temporary table before they're moved over to `system.jobs`.

Reply:
In that case, we could batch this in separate transactions if we were worried about the transaction being too big. It doesn't seem that we're super concerned about that. Is there a strong need for atomicity here updating the jobs table and other things happening in this txn?

Reply:
> Is there a strong need for atomicity here updating the jobs table and other things happening in this txn?

This txn updates the job state to indicate that the system tables have been restored (`details.SystemTablesRestored`). This information is used to protect against the case where:
- The job migrates the temporary system table data
- The job restarts and retries to re-restore the temporary system table's data, but on restore AddSSTable will now shadow some keys with a different value and throws an error.

I don't think that there is any restriction that requires this update to be in the same transaction as the updating the `system.jobs` table since we don't allow stale reads.

That being said, this change can be simplified by splitting the system table restoration step into 2 steps:
1. Updating/migrating the system data in the staging table. Even if this phase is idempotent, we cannot re-attempt to restore data over this migrated data, so we need to persist state in the job keeping track of migrations.
2. Actually moving the data over to the live system tables. Previously, this may have not been idempotent and we used the previously mentioned state on the job to check if we've restored this system table, but I now realize that this can be made to be idempotent by upserting the values, rather than inserting them. Perhaps enforcing that these are idempotent is the way to go.

Reply:
> but I now realize that this can be made to be idempotent by upserting the values, rather than inserting them. Perhaps enforcing that these are idempotent is the way to go.

Rather than `UPSERT` it feels like what you want might be `INSERT ... ON CONFLICT DO NOTHING`. That way if the job has already been adopted we don't muck with it.

Reply:
Did a bit of the refactoring mentioned here. If I add batching, I think that could help with the number of roundtrips. Will tackle that next.